### PR TITLE
Add InteractionController decorator

### PR DESCRIPTION
**Pull request by Booster1212**

### CODE REVIEW

File: src/core/server/decorators/interaction.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import alt, { IVector3 } from 'alt-server';
+import { Athena } from '../api/athena';
+
+export function interaction(
```
Comment (Stuyk) [inline src/core/server/decorators/interaction.ts:4]: Provide an example and document this specific new function / decorator with an example.

**Review by Stuyk — CHANGES_REQUESTED** [review]

File: src/core/server/decorators/interaction.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+/**
+ * @param {IVector3} pos - The position of the interaction.
+ * @param {string} description - The description of the interaction.
+ * @param {number} range - The range of the interaction.
+ * @param {number} dim - The dimension to search in.
+ * @param {string} uid - The uid of the interaction.
+ * Decorator for Athena's Interaction Controller
+ * ```
+ * class ExampleDecorators {
```
Comment (xLuxy) [inline src/core/server/decorators/interaction.ts:12]: Should be `ExampleInteractions` - decorators implies that you have more than one decorator tbh.

File: src/core/server/decorators/interaction.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+ *   }
+ * }
+ * ```
+ */
+
+export function interaction(
+    pos: IVector3,
+    description: string,
+    range: number,
+    dim: number,
```
Comment (xLuxy) [inline src/core/server/decorators/interaction.ts:25]: should be `position`

File: src/core/server/decorators/interaction.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+ *   @interaction({ x: 0, y: 0, z: 0 }, 'Hello World!', 25, 0, 'UUID')
+ *   static exampleInteraction() {
+ *      alt.log("Interaction controller got triggerd by player through decorator!");
+ *   }
+ * }
+ * ```
+ */
+
+export function interaction(
+    pos: IVector3,
```
Comment (xLuxy) [inline src/core/server/decorators/interaction.ts:22]: should be `dimension`

**Review by xLuxy — CHANGES_REQUESTED** [review]

File: src/core/server/decorators/interaction.ts [hunk 5]
```diff
@@ -0,0 +1,45 @@
+import alt, { IVector3 } from 'alt-server';
+import { Athena } from '../api/athena';
+
+/**
+ * @param {IVector3} pos - The position of the interaction.
+ * @param {string} description - The description of the interaction.
+ * @param {number} range - The range of the interaction.
+ * @param {number} dimension - The dimension to search in.
+ * @param {string} uid - The uid of the interaction.
+ * Decorator for Athena's Interaction Controller
```
Comment (xLuxy) [inline src/core/server/decorators/interaction.ts:10]: Should be above the parameters

**Review by xLuxy — CHANGES_REQUESTED** [review]

File: src/core/server/decorators/interaction.ts [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+        Athena.controllers.interaction.add({
+            position: position,
+            description: description,
+            dimension: dimension,
+            isPlayerOnly: playerOnly,
+            isVehicleOnly: vehicleOnly,
+            range: range,
+            debug: debug,
+            uid: uid,
+            callback: callback,
```
Comment (LeonardSSH) [inline src/core/server/decorators/interaction.ts:42]: Use object property-value shorthand.

```suggestion
            position,
            description,
            dimension,
            range,
            debug,
            uid,
            callback,
            isPlayerOnly: playerOnly,
            isVehicleOnly: vehicleOnly,
```

**Review by LeonardSSH — CHANGES_REQUESTED** [review]

File: src/core/server/decorators/interaction.ts [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+ *      alt.log("Interaction controller got triggerd by player through decorator!");
+ *   }
+ * }
+ * ```
+ * @param {IVector3} position - The position of the interaction.
+ * @param {string} description - The description of the interaction.
+ * @param {number} range - The range of the interaction.
+ * @param {number} dimension - The dimension to search in.
+ * @param {string} uid - The uid of the interaction.
+ */
```
Comment (LeonardSSH) [inline src/core/server/decorators/interaction.ts:19]: Docs for `debug`, `playerOnly` and `vehicleOnly`?

Comment (LeonardSSH) [reply src/core/server/decorators/interaction.ts:19]: Even if they are optional, I think they should have docs.

File: src/core/server/decorators/interaction.ts [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+ * @param {string} data.description - The description of the interaction.
+ * @param {number} data.range - The range of the interaction.
+ * @param {number} data.dimension - The dimension to search in.
+ * @param {string} data.uid - The uid of the interaction.
+ * @param {boolean} data.debug - If the interaction should be debugged.
+ * @param {boolean} data.playerOnly - If the interaction should only be triggered by players.
+ * @param {boolean} data.vehicleOnly - If the interaction should only be triggered by vehicles.
+ */
+export function interaction(data: Interaction) {
+    alt.log(data);
```
Comment (LeonardSSH) [inline src/core/server/decorators/interaction.ts:54]:
```suggestion

```

File: src/core/server/decorators/interaction.ts [hunk 9]
```diff
@@ -0,0 +1,37 @@
+import alt from 'alt-server';
+import { Interaction } from '../../shared/interfaces/interaction';
+import { Athena } from '../api/athena';
+/**
+ * Decorator for Athena's Interaction Controller
```
Comment (LeonardSSH) [inline src/core/server/decorators/interaction.ts:5]: A little space here.

```suggestion
import { Athena } from '../api/athena';

/**
 * Decorator for Athena's Interaction Controller
```

**Review by LeonardSSH — CHANGES_REQUESTED** [review]

**Pull request merged**